rfq/order.go

```diff
@@ -297,6 +297,12 @@ func (c *AssetPurchasePolicy) CheckHtlcCompliance(
 		return fmt.Errorf("error summing asset balance: %w", err)
 	}
 
+	// Due to rounding errors, we may slightly underreport the incoming
+	// value of the asset. So we increase it by exactly one asset unit to
+	// ensure that the fee logic in lnd does not reject the HTLC.
+	roundingCorrection := rfqmath.NewBigIntFromUint64(1)
+	assetAmt = assetAmt.Add(roundingCorrection)
```

Comment:
Would adding an entire asset unit be acceptable in the scenario where where one asset unit is a considerable amount / value to the user? e.g. the amount represented an 1e8 sats.  Or are the rounding errors only expected when the values are de minis?

What other methods might be available to address the source of the rounding errors?

Reply:
just to clarify: this PR doesn't "fix" any rounding related code. It just adjusts a check to allow an HTLC to be forwarded that is affected by the rounding down. The rounding down will always happen, due to the integer nature of taproot assets (see [RFQ/decimal display doc](https://github.com/lightninglabs/taproot-assets/blob/main/docs/rfq-and-decimal-display.md) for more info).
The decision we can make is: do we force the sender to over pay by one unit so that the ultimate receiver always sees the expected (or even one unit more) balance? Or do we allow under payment of one unit, leading to the scenario where the ultimate receiver sees a final balance lower than expected.

But since this only fixes a _check_, I don't think we should block this PR and look at the decision holistically.